pkg/obfuscator/regex_test.go

```diff
@@ -0,0 +1,43 @@
+package obfuscator
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRegexObfuscator(t *testing.T) {
+	for _, tc := range []struct {
+		name    string
+		pattern string
+		input   string
+		output  string
+	}{
+		{
+			name:    "match-basic",
+			pattern: `(super\-)+(secret)`,
+			input:   "line with a super-super-secret, super-secret and a non-secret",
+			output:  "line with a xxxxxxxxxxxxxxxxxx, xxxxxxxxxxxx and a non-secret",
+		},
+		{
+			name:    "match-everything",
+			pattern: `.*`,
+			input:   "line with a super-super-secret, super-secret and a non-secret",
+			output:  "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
+		},
+		{
+			name:    "match-nothing",
+			pattern: `(super\-)+(secret)`,
+			input:   "no secrets here",
+			output:  "no secrets here",
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			o, err := NewRegexObfuscator(tc.pattern, "All")
```

Comment:
ObfuscateTargetAll please, do you also want to test the other types individually?